File: Busca/Implementacao/Javascript/linear_search.js
// Pesquisa Linear (Linear Search)

function linear_search(arr, target){
    for(let i=0; i < arr.length; i++){
        if(arr[i] === target){
            return i // Retorna o índice do elemento caso seja encontrado.
        };
    };
    return -1 // Retona -1 se o elemento não for encontado.
};

// Exemplo de Execução:

let arr = [10, 20, 30, 40, 50];
let target = 30;
let result = linear_search(arr, target);
console.log(result !== -1 ? `Elemento encontrado no índice: ${result}` : `Elemento não encontrado`);

File: Busca/Implementacao/Python/linear_search.py

# Pesquisa Linear (Linear Search)

def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i ## Retorna o índice caso o elemento seja encontrado
    return -1 ## Retorna -1 se elemento não for encontrado

#Exemplo de Execução
arr = [10, 20, 30, 40, 50]
target = 30
result = linear_search(arr, target)
print(f"Elemento encontrado no índice: {result}" if result != -1 else "Elemento não encontrado")